src/System.Management.Automation/engine/hostifaces/PowerShellProcessInstance.cs

```diff
@@ -50,7 +52,26 @@ static PowerShellProcessInstance()
         /// <param name="workingDirectory">Specifies the initial working directory for the new powershell process.</param>
         public PowerShellProcessInstance(Version powerShellVersion, PSCredential credential, ScriptBlock initializationScript, bool useWow64, string workingDirectory)
         {
-            string processArguments = " -s -NoLogo -NoProfile";
+            string exePath = PwshExePath;
+            string processArguments = String.Empty;
+
+#if !UNIX
+            // if requested PS version was "5.1" then we start Windows PS instead of PS Core
+            // Version parameter needs to go before all other parameters.
+            if (powerShellVersion != null)
+            {
+                if ((powerShellVersion.Major == 5) && (powerShellVersion.Minor == 1))
+                {
+                    exePath = WinPwshExePath;
+                    processArguments = "-Version 5.1";
+                }
+            }
+#endif
+
+            processArguments = string.Format(
```

Comment:
I would reverse the order here and tack on '-Version 5.1' to common '-s -NoLogo ...' arguments, only when requested.

Reply:
It was done this way because in Windows PS code in the same spot I found following comment (and Windows PS code fills in `-Version` before anything else):
```
            // Version parameter needs to go before all other parameters because the native layer looks for Version or 
            // PSConsoleFile parameters before parsing other parameters.
            // The other parameters get parsed in the managed layer.
```